src/devana/syntax_abstraction/functioninfo.py

```diff
@@ -10,92 +10,238 @@
+    CONST = ModificationKind.CONST
+    EXPLICIT = ModificationKind.EXPLICIT
+    STATIC = ModificationKind.STATIC
+    VIRTUAL = ModificationKind.VIRTUAL
+    PURE_VIRTUAL = ModificationKind.PURE_VIRTUAL
+    INLINE = ModificationKind.INLINE
+    FINAL = ModificationKind.FINAL
+    OVERRIDE = ModificationKind.OVERRIDE
+    DELETE = ModificationKind.DELETE
+    DEFAULT = ModificationKind.DEFAULT
+    CONSTEXPR = ModificationKind.CONSTEXPR
+    VOLATILE = ModificationKind.VOLATILE
+    NOEXCEPT = ModificationKind.NOEXCEPT
+
+    def __init__(self, value: Optional[int] = None, expect_value: Optional[str] = None):
+        self._noexcept_value: Optional[str] = None
+        if value is not None:
+            self._value = FunctionModification.ModificationKind(value)
+            if value & FunctionModification.ModificationKind.NOEXCEPT:
+                if expect_value is not None:
+                    self._noexcept_value = expect_value
+                else:
+                    self._noexcept_value = None
+        else:
+            self._value = FunctionModification.ModificationKind.NONE
+
+    @classmethod
+    def create_noexcept(cls, noexcept_value):
+        return TypeModification.POINTER(noexcept_value)  # noqa pylint: disable=not-callable
+
+    @property
+    def value(self) -> ModificationKind:
+        return self._value
+
+    def __and__(self, other):
+        if isinstance(other, type(self)):
+            result = FunctionModification(self.value & other.value)
+            if result.is_noexcept:
+                if other.noexcept_value is not None:
+                    if self.noexcept_value is None:
+                        result.noexcept_value = other.noexcept_value
+                    else:
+                        if self.noexcept_value != other.noexcept_value:
+                            result.noexcept_value = None
+                else:
+                    result.noexcept_value = self.noexcept_value
+            return result
+        elif isinstance(other, FunctionModification.ModificationKind):
+            result = FunctionModification(self.value & other)
+            # I do not know if it's required here
+            # ↓ ↓ ↓ ↓ ↓ ↓ ↓ ↓ ↓ ↓ ↓ ↓ ↓ ↓ ↓ ↓ ↓ ↓
+            # if result.is_noexcept:
+            #     if self.is_noexcept and other == FunctionModification.ModificationKind.NOEXCEPT \
+            #             and self.noexcept_value is None:
+            #         result._noexcept_value = None
+            return result
+        raise NotImplementedError()
+
+    def __or__(self, other):
+        if isinstance(other, type(self)):
+            result = FunctionModification(self.value | other.value)
+            if result.is_noexcept:
+                if other.noexcept_value is not None:
+                    if self.noexcept_value is None:
+                        result.noexcept_value = other.noexcept_value
+                    else:
+                        pass  # What to do here
```

Comment:
1. noexcept_value NIL for both -> just keep it
2. noexcept_value NIL for one and for second value -> force value
3. noexcept_value both no-NIL -> raise exception